app/scripts/controllers/detect-tokens.js

```diff
@@ -247,7 +242,8 @@ export default class DetectTokensController extends StaticIntervalPollingControl
    */
   restartTokenDetection({ selectedAddress, chainId } = {}) {
     const addressAgainstWhichToDetect = selectedAddress ?? this.selectedAddress;
-    const chainIdAgainstWhichToDetect = chainId ?? this.chainId;
+    const chainIdAgainstWhichToDetect =
+      chainId ?? this.getChainIdFromNetworkStore();
```

Comment:
During a network switch, `this.chainId` is from the old chain until `NetworkController:stateChange` fires and updates it. But `TokenListController:stateChange` can fire first, leaving it stale at this point.  This change gets the updated chain id from the network controller, as happens elsewhere in this file.